src/args.rs

```diff
@@ -23,7 +26,8 @@ use crate::{
 
 const ALL_NODES_QUERY: &str = "(_) @node";
 
-#[derive(Parser)]
+#[derive(Builder, Clone, Default, Parser)]
```

Comment:
Exposed an `ArgsBuilder` for `tree-sitter-lint` to use (vs building `Args` clunkily from passing "command-line args")

Presumably there should be some additional validation eg that some of the invariants that `clap` checks when parsing command-line args are not violated when using the builder, but I haven't done that yet